outbound.js

```diff
@@ -312,13 +320,41 @@ exports._add_file = function (hmail) {
     }
 };
 
-exports._list_file = function (file) {
-    // TODO: output more data here?
-    console.log("Q: " + file);
+exports._list_file = function (file, cb) {
+    var tl_reader = fs.createReadStream(path.join(queue_dir, file), {start: 0, end: 3});
+    tl_reader.on('error', function (err) {
+        console.error("Error reading queue file: " + file + ":", err);
+    });
+    tl_reader.once('data', function (buf) {
+        // I'm making the assumption here we won't ever read less than 4 bytes
+        // as no filesystem on the planet should be that dumb...
+        tl_reader.destroy();
+        var todo_len = (buf[0] << 24) + (buf[1] << 16) + (buf[2] << 8) + buf[3];
+        var td_reader = fs.createReadStream(path.join(queue_dir, file), {encoding: 'utf8', start: 4, end: todo_len + 3});
+        var todo = '';
+        td_reader.on('data', function (str) {
+            todo += str;
+            if (Buffer.byteLength(todo) === todo_len) {
+                // we read everything
+                var todo_struct = JSON.parse(todo);
+                todo_struct.rcpt_to = todo_struct.rcpt_to.map(function (a) { return new Address (a); });
+                todo_struct.mail_from = new Address (todo_struct.mail_from);
+                todo_struct.file = file;
+                todo_struct.full_path = path.join(queue_dir, file);
+                cb(null, todo_struct);
+            }
+        });
+        td_reader.on('end', function () {
+            if (Buffer.byteLength(todo) !== todo_len) {
+                console.error("Didn't find right amount of data in todo for file:", file);
```

Comment:
If I did that the whole thing would stop (but I guess now it would hang).

> On May 10, 2016, at 7:19 PM, Steve Freegard <notifications@github.com> wrote:
> 
> In outbound.js:
> 
> > +        var todo = '';
> > +        td_reader.on('data', function (str) {
> > +            todo += str;
> > +            if (Buffer.byteLength(todo) === todo_len) {
> > +                // we read everything
> > +                var todo_struct = JSON.parse(todo);
> > +                todo_struct.rcpt_to = todo_struct.rcpt_to.map(function (a) { return new Address (a); });
> > +                todo_struct.mail_from = new Address (todo_struct.mail_from);
> > +                todo_struct.file = file;
> > +                todo_struct.full_path = path.join(queue_dir, file);
> > +                cb(null, todo_struct);
> > +            }
> > +        });
> > +        td_reader.on('end', function () {
> > +            if (Buffer.byteLength(todo) !== todo_len) {
> > +                console.error("Didn't find right amount of data in todo for file:", file);
> Shouldn't this be return cb(new Error("Didn't find right amount of data in todo for file: ", file) instead?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>